federation/cmd/federation-apiserver/app/core.go

```diff
@@ -18,6 +18,14 @@ package app
 
 import (
 	"github.com/golang/glog"
+
+	// HACK to ensure that rest mapper from pkg/api is registered for groupName="".
+	// This is required because both pkg/api/install and federation/apis/core/install
+	// are installing their respective groupMeta at the same groupName.
+	// federation/apis/core/install has only a subset of resources and hence if it gets registered first, then installation of v1 API fails in pkg/master.
+	// TODO(nikhiljindal): Fix this by ensuring that pkg/api/install and federation/apis/core/install do not conflict with each other.
+	_ "k8s.io/kubernetes/pkg/api/install"
```

Comment:
> I think this will break the swagger spec that federation puts out? Which may not be a concern?

federation-apiserver was ending up importing pkg/api/install even before this change. Here is the dependency chain::
```
federation/cmd/federation-apiserver/app/server.go
federation/cmd/federation-apiserver/app/options/options.go
pkg/genericapiserver/options/server_run_options.go
pkg/admission/types.go
pkg/client/clientset_generated/internalclientset/import_known_versions.go
pkg/api/install/install.go
```




> note that the rest mapper is hard coded to get added to api.Scheme...

The particular problem here is the registered package. Whichever install gets called first, calls registered.RegisterGroup() with its own groupMeta.
Later, in pkg/master,  when registered.GroupOrDie() is called from InstallAPIGroups(), it returns the groupMeta which was registered first.

I am planning to fix it by updating the install packages in all apis so that pkg/api and federation/apis/core/install pass different keys to register and then retrieve their respective groupMetas.